synapse/module_api/__init__.py

```diff
@@ -43,6 +92,9 @@ def __init__(self, hs, auth_handler):
         self._auth = hs.get_auth()
         self._auth_handler = auth_handler
 
+        self.http_client = hs.get_simple_http_client()  # type: SimpleHttpClient
+        self.public_room_list_manager = PublicRoomListManager(hs)
```

Comment:
I would recommend exposing these as `@property` methods, so that they can have some docstring describing what they do

Reply:
I have done so in https://github.com/matrix-org/synapse/pull/8292/commits/5730d17ac71cb23b3e9407e1bd591071608d7b19, though I'm not sure if the contents of the docstrings were entirely what you were looking for.

Reply:
they look good to me